### PR TITLE
AI Review updates for Example PSE & PSE 1

### PR DESCRIPTION
**Pull request by kelsey-steven-ada**
[Asana Ticket](https://app.asana.com/1/181459410160484/project/1205655776549324/task/1210346586413447?focus=true)
[Parent ticket with further info](https://app.asana.com/1/181459410160484/project/1205655776549324/task/1209973030564733?focus=true)
[Planning doc from team discussion](https://docs.google.com/document/d/1yk_5jP7Pz__o_Ba-iUzPRA5Cvp3pjGdpAGBjERf6ibQ/edit?tab=t.0) (new tabs added for organizing updates)

Changes
- Adds prompt templates for reviewing clarifying questions and logical steps to repo under `shared-resources` folder
- Updates "About PSEs" to mention new first step of gathering information in list of what to expect
- Updates existing prompt text for "Clarifying Questions" and "Logical Steps" to better reflect our expectations for students
- Updates example responses for Clarifying Questions to steer further away from validation questions to better match our goals for clarifying questions
- Fixes inconsistent capitalization in sub bullets of both logical steps examples
- Adds 3 new fields to the example PSE & PSE 1 along with hints where relevant and example responses:
    - "Gather Information"
    - "Review Clarifying Questions"
    - "Review logical steps"

### CODE REVIEW

File: 01/highest-rated.md [hunk 1]
```diff
@@ -42,19 +103,98 @@ Consider the following for inspiration:
...
+While we build our skills in breaking down a problem and choosing clarifying questions, let’s use an external tool like ChatGPT to review the questions we wrote above. Our goals are to: 
+- ensure we are asking questions that will tell us new information about the problem space
+- check our understanding of the information we expect to get from those questions
+- uncover other questions that could be useful to ask
+- understand why those other questions could be helpful
+
+<br>
+
+For this question we will:
+1. Build a prompt using [the template linked here](../shared-resources/questions-review-prompt-template.md)
```
Comment (yangashley) [inline 01/highest-rated.md:141]: When I click this link it opens the file as unformatted markdown. Is that what we're expecting?

image

If not, I haven't figured out why the file isn't opening up as formatted markdown.

Comment (kelsey-steven-ada) [reply 01/highest-rated.md:141]: Hmm, I was hoping it would render the formatting, but the only actual markdown is the title "Template for ...", so I'm not sure it would be an issue for students. We could always link to the file in GitHub rather than linking to the file in Learn if that's a concern.

File: about-pses/example-pse.md [hunk 2]
```diff
@@ -58,20 +130,24 @@ Try not to pose questions that are too specific to the implementation. For examp
 
 As stated, this list contains several questions that overlap in what they are asking as an example of different perspectives. Try not to include multiple questions about the same topic in your own questions. For example, if we ask about how to handle invalid input, we only need to ask once, not in three different ways as we have here, especially since the sample answers here are contradictory!
```
Comment (kelsey-steven-ada) [inline about-pses/example-pse.md:131]: Were you thinking something like italics, or breaking up the paragraph some? A couple options:

```suggestion
As stated, this list contains several questions that overlap in what they are asking as an example of different perspectives. _Try not to include multiple questions about the same topic in your own questions._ For example, if we ask about how to handle invalid input, we only need to ask once, not in three different ways as we have here, especially since the sample answers here are contradictory!
```

```suggestion
As stated, this list contains several questions that overlap in what they are asking as an example of different perspectives. Try not to include multiple questions about the same topic in your own questions. 
- For example, if we ask about how to handle invalid input, we only need to ask once, not in three different ways as we have here, especially since the sample answers here are contradictory!
```

File: about-pses/example-pse.md [hunk 3]
```diff
@@ -58,20 +130,24 @@ Try not to pose questions that are too specific to the implementation. For examp
...
+    - I want to know if my implementation needs to be flexible enough to extend it with additional game rules.
+    - While I could write a helper function that accepts a data structure that encodes the valid moves and what beats what, I'll assume that I should focus on the basic rules for now. This way, I can get a working solution faster, and if I need to extend the game later, I can refactor my code to be more flexible.
+6. **When friends use rock, paper, scissors to make decisions, there are often more than two players. Should I keep my approach focused on the basic rules for 2 players?**
+    - I want to know if I should I use an implementation that could be extended to support multiple players simultaneously.
+    - While I could write a helper function that accepts a list of moves made by all players, the multiplayer version introduces a few complications into resolving a round that I'd like to avoid for now. I'll assume that I should focus on the basic rules for now. This way, I can get a working solution faster, and if I need to extend the game later, I can refactor my code to be more flexible.
```
Comment (kelsey-steven-ada) [inline about-pses/example-pse.md:150]: I like that, it feels a little less vague and focuses in on where we want students putting effort. How does this sound to incorporate that language?

```suggestion
    - While I could write a helper function that accepts a list of moves made by all players, the multiplayer version increases the scope of the problem. I'll assume that I should focus on the basic rules for now. This way, I can get a working solution faster, and if I need to extend the game later, I can refactor my code to be more flexible.
```

File: about-pses/example-pse.md (hunk 2 above)
Comment (yangashley) [reply about-pses/example-pse.md:131]: I think breaking it into chunks and using the bolding could be helpful because visually it is a long block of text.

File: about-pses/example-pse.md (hunk 3 above)
Comment (yangashley) [reply about-pses/example-pse.md:150]: Yeah! That looks good to me.

File: about-pses/example-pse.md (hunk 2 above)
Comment (kelsey-steven-ada) [inline about-pses/example-pse.md:131]:
```suggestion
As stated, this list contains several questions that overlap in what they are asking as an example of different perspectives. *Try not to include multiple questions about the same topic in your own questions.*
- For example, if we ask about how to handle invalid input, we only need to ask once, not in three different ways as we have here, especially since the sample answers here are contradictory!
```

File: about-pses/example-pse.md [hunk 4]
```diff
@@ -85,8 +162,87 @@ Even if we make a decision about how to handle a particular issue here, the unit
...
+1. Build a prompt using [the template linked here](../shared-resources/questions-review-prompt-template.md)
+2. Share the prompt with an AI tool like ChatGPT
+3. Hold a discussion with the tool to ensure our understanding of what new information we would get from the initial questions submitted, and what other questions we might want to ask.
+4. Reflect on the information shared by the AI tool and summarize its findings
+
+<br>
+
+In the box below, please submit:
+1. A shareable link to your conversation in ChatGPT 
+    - [Documentation for creating a shareable link in ChatGPT](https://help.openai.com/en/articles/7925741-chatgpt-shared-links-faq)
```
[the remark this answers is not in the document]
Comment (anselrognlie) [reply about-pses/example-pse.md:191]: What if we included some potential follow-up questions, like "How did you select on what to base your analysis of my questions?" Like I'm wondering whether there's a way we can get Chat to spit out more general principles that students could then apply to other PSEs. There's _some_ of that in the "why ask this" and "what does it clarify" bits it tacks on, but can we model how students can extract that more general knowledge directly?

File: 01/highest-rated.md
Comment (anselrognlie) [inline 01/highest-rated.md:1]: Just acknowledging that I didn't really look at this file. I focused on the about examples as the place to hang my thoughts. I figured whatever decisions we make there would be carried through the rest of the materials as well.

File: about-pses/example-pse.md [hunk 5]
```diff
@@ -85,8 +162,87 @@ Even if we make a decision about how to handle a particular issue here, the unit
...
+While we build our skills in breaking down a problem and choosing clarifying questions, let’s use an external tool like ChatGPT to review the questions we wrote above. Our goals are to: 
+- ensure we are asking questions that will tell us new information about the problem space
+- check our understanding of the information we expect to get from those questions
+- uncover other questions that could be useful to ask
+- understand why those other questions could be helpful
+
+<br>
+
+For this question we will:
+1. Build a prompt using [the template linked here](../shared-resources/questions-review-prompt-template.md)
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply about-pses/example-pse.md:182]: I'd love to get other folk's perspective on this. We can distribute the templates, but then would we not link anything here and have a line telling them to go get it at X place? Even when we provide a resource elsewhere, our process is generally to link a resource where it is used for ease of access. 

I know it is deeply disappointing to lose work, and also, I think students should extremely quickly be very comfortable opening things in another tab.

File: about-pses/example-pse.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply about-pses/example-pse.md:191]: I think it would be really helpful to add some follow up questions to better show how it could be a conversation and how to dig into the understanding deeper.

File: about-pses/example-pse.md [hunk 6]
```diff
@@ -85,8 +162,87 @@ Even if we make a decision about how to handle a particular issue here, the unit
...
+    - I want to know whether the function is intended to be flexible and forgiving with missing inputs, or if both arguments are strictly required.
+```
+
+<br>
+
+Depending on exactly what ChatGPT shares, a reflection and summary might look like:
+
+<br>
+
+Chat link: https://chatgpt.com/share/682fa17f-b1a8-8005-9dec-301093934696
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply about-pses/example-pse.md:232]: Something like an internal .md link would be more reliable. One of the things I wanted to ask the team is if we want to include links to a sample conversation in our sample response or just include a placeholder like `<url to your chat>`.

File: about-pses/example-pse.md [hunk 7]
```diff
@@ -189,30 +343,32 @@ def test_invalid_input():
...
 ### !challenge
-
 * type: paragraph
 * id: 2bf3b413-e220-46aa-a99a-6c5f3d9715a9
 * title: Create Logical Steps
 * topics: pse
-
 ##### !question
 
 Without writing code, describe how you would implement `winner` in enough detail that someone else could write the code.
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply about-pses/example-pse.md:354]: I agree that's an easy place to get caught up and think you need to account for each line. What about something like:

```suggestion
Without writing code, describe how you would implement `winner` in enough detail that an experienced developer could reasonably write an implementation based on the description. We should capture the main use cases, but the steps do not need to be a detailed plan for every contingency. The objective is to create a roadmap that we can use to keep ourselves oriented towards our goal, it is okay to leave some of the finer details to be worked out in the implementation itself!
```

File: shared-resources/steps-review-prompt-template.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+```
+
+Examples:
+```
+
+```
+
+I would like assistance reviewing the steps that I came up with to approach the coding problem above. In your response, do not include any code to solve the problem. Assume that input validation is handled ahead of this function unless it is explicitly mentioned in the problem statement or examples, the focus should be on the core of the problem. If my approach is missing steps or scenarios, include information about what is missing and why that step is important. I would also like help evaluating: 
+- if the steps make sense for the problem being solved
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply shared-resources/steps-review-prompt-template.md:19]:
```suggestion
- if the steps make sense for the problem being solved  
- if I provide enough detail in each step that I show that I understand why the step is important and how it works in the context of the overall solution
```

File: about-pses/about-pses.md [hunk 9]
```diff
@@ -22,11 +22,12 @@ Problem-Solving Exercise assignments (PSEs) are assignments to practice problem-
...
-    * iii. Break down the problem into logical steps.   
-1. A code challenge to solve the problem
+2. A series of prompts to guide your understanding of the problem and scaffold our solution:
+    1. Describe your understanding of the problem requirements based on the problem statement and examples
+    2. Ask at least three clarifying questions that would give you new information about the problem
+    3. Write example inputs/outputs and unit tests for at least one nominal and one edge case  
+    4. Break down the problem into logical steps
+3. A code challenge to solve the problem
 
 ### Learning Goals
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply about-pses/about-pses.md:29]: Updated, resolving to remove old comments before new reviews.

File: about-pses/example-pse.md [hunk 10]
```diff
@@ -29,7 +30,63 @@ Create a function named `winner` that takes two arguments, the move for `player_
...
+- What assumptions can we make based on the information in the problem statement? 
+- What further information do the example inputs and outputs give us?
+
+##### !end-question
+##### !hint
+
+Further questions to ask as you read through the problem statement and examples:
+- What is the goal of the function?
+- What are the types of the expected inputs and outputs?
+- Are there any restrictions on any of the inputs?
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply about-pses/example-pse.md:62]: This has been added, resolving before new reviews

File: about-pses/about-pses.md [hunk 11]
```diff
@@ -3,38 +3,35 @@
...
 - Solving small coding problems while working towards a bigger solution
 
 ## What Are PSEs?
 
-Problem-Solving Exercise assignments (PSEs) are assignments to practice problem-solving skills.
+Problem-Solving Exercise assignments (PSEs) are assignments to practice problem-solving skills that we will use in interviews and when communicating about code with others.
 
-The exercises will touch on problem-solving skills on coding problems. The format has three parts:
+Interviews are conducted in a setting that enables discussion and thinking out loud. As a result, the practice in PSEs will be an imperfect approximation, but the act of thinking about and rehearsing how we can word these ideas is valuable preparation. In these exercises, we will write down what would mostly be talked through in an actual interview. 
+- During an interview, writing some of the discussion points down can still be useful but our notes there would likely not be to the same level of detail as the responses for PSEs.
```
Comment (apradoada) [inline about-pses/about-pses.md:19]: I feel like this might be a good time to mention something along the lines of us asking them to interact with AI tools during the PSE process but that they will likely not have access to those tools in an interview.
```suggestion
- During an interview, writing some of the discussion points down can still be useful but our notes there would likely not be to the same level of detail as the responses for PSEs.
    - For example, our PSEs will ask you to use AI tools to supplement your critical thinking . You will likely not have access to AI tools in an Interview.
```

File: about-pses/example-pse.md [hunk 12]
```diff
@@ -29,64 +29,189 @@ Create a function named `winner` that takes two arguments, the move for `player_
...
-##### !end-question
+<br>
+
+In the box below, list 5 or more observations about the problem or questions whose answers would clarify the problem statement. For each observation or question, include information on why that observation is important or why you are asking the question
+- For each observation, answer how that observation will affect your approach to the problem
+- For each question, describe what you are hoping to clarify about the problem and provide an answer which includes the effect your decision would have on how you might approach the problem.
+
+<br>
 
+As you come up with observations and questions, assume that error handling for invalid data is managed outside the function. We want to focus on the core behavior of the function we will write.
```
Comment (apradoada) [inline about-pses/example-pse.md:54]: Repeat from comment on `highest-rated.md`

```suggestion
As you come up with observations and questions, assume that error handling for invalid data is managed outside the function. We want to focus on the core behavior of the function we will write. Examples of invalid data we don't need to manage may include:
- Mismatched data types: If a function expects a string, you may assume it will receive a string. This follows for nested data as well.
- String Sanitization: You may assume strings will not include invalid characters.
```

File: about-pses/example-pse.md [hunk 13]
```diff
@@ -29,64 +29,189 @@ Create a function named `winner` that takes two arguments, the move for `player_
...
+##### !hint
+
+Further questions to ask as you read through the problem statement and examples:
+- What is the goal of the function?
+- What are the types of the expected inputs and outputs?
+- Are there any restrictions on any of the inputs?
+  - For example: if any of the inputs are a list, do we know anything about how the list is ordered?
+- What do the examples show us about the data types and values that are allowed for our inputs?
+- What do the examples tell us about the return value in different scenarios?
+- Reflect on the observations you have made so far, what questions would give you new information?
```
Comment (apradoada) [inline about-pses/example-pse.md:66]: Repeat comment from `highest-rated.md`

```suggestion
- Reflecting on the observations you have made so far, what questions would give you new information?
```

File: 01/highest-rated.md [hunk 14]
```diff
@@ -20,41 +20,156 @@ This function should take in a list of dictionaries named `restaurants` as a par
...
-Consider the following for inspiration:
+Further questions to ask as you read through the problem statement and examples:
+- What is the goal of the function?
+- What are the types of the expected inputs and outputs?
+- Are there any restrictions on any of the inputs?
+  - For example: if any of the inputs are a list, do we know anything about how the list is ordered?
+- What do the examples show us about the data types and values that are allowed for our inputs?
+- What do the examples tell us about the return value in different scenarios?
+- Reflect on the observations you have made so far, what questions would give you new information?
+- Assume error handling for invalid data is managed outside the function.
```
Comment (apradoada) [inline 01/highest-rated.md:58]: If we do bring this up earlier, I wonder if we need it again here? It also seems a little bit out of place to me because it is the only one that isn't phrased as a question.

```suggestion
```

File: about-pses/example-pse.md [hunk 15]
```diff
@@ -29,64 +29,189 @@ Create a function named `winner` that takes two arguments, the move for `player_
...
+- Assume error handling for invalid data is managed outside the function.
```
Comment (apradoada) [inline about-pses/example-pse.md:67]: Repeated comment from `highest-rated.md`
```suggestion
```

File: about-pses/example-pse.md [hunk 16]
```diff
@@ -29,64 +29,189 @@ Create a function named `winner` that takes two arguments, the move for `player_
...
 
-Even if we make a decision about how to handle a particular issue here, the unit tests may lead us to reconsider that decision. For example, if we decide to return `None` for invalid input, but the unit tests expect an error to be raised, we may need to change our implementation, but there's no need to come back and update the clarifying questions. The important thing is to think through the problem and make a decision based on the information we have at the time. When the tests differ from our initial thoughts, we can think of that as a case where the interviewer has given us new information that we need to incorporate into our solution.
+One of many possible sample responses could look like:
+
+<br>
+
+1. I need to implement a function called `winner` that takes two arguments: `player_1` and `player_2`.
+    - This means I need to create a function definition with two parameters named exactly `player_1` and `player_2`.
+
+2. Each argument represents a move for player_1 or player_2 and will be one of "rock", "paper", or "scissors".
```
Comment (apradoada) [inline about-pses/example-pse.md:99]:
```suggestion
2. Each argument represents a move for `player_1` or `player_2` and will be one of "rock", "paper", or "scissors".
```

File: about-pses/example-pse.md [hunk 17]
```diff
@@ -29,64 +29,189 @@ Create a function named `winner` that takes two arguments, the move for `player_
...
+5. The table of rules contains text "input (first argument)" and "input (second argument)"
+    - This tells me when deciding who is the winner, I need to be mindful that the first argument's value always represents player_1 and the second argument represents player_2.
+    - This is supported by the first example where `player_1 = "rock"` and `player_2 = "scissors"`, and the result was `"Player 1 wins!"`.
+
+6. Based on the table of rules: "rock" beats "scissors", "scissors" beats "paper", "paper" beats "rock".
+    - When developing my solution, I will use these rules to guide the conditional statements I need to write to determine the winner.
+
+<br>
+
+We should keep in mind that sometimes, areas of clarification may not jump out at us when we first read the prompt. It's completely OK to think through the prompt, and even start working on it. As we start to write code, we may note something we missed before or realize that we need to ask a clarifying question. This is a normal part of the problem-solving process. So don't feel like we need to note everything or ask all our clarifying questions before we start working on the problem.
```
Comment (apradoada) [inline about-pses/example-pse.md:118]:
```suggestion
We should keep in mind that sometimes, areas of clarification may not jump out at us when we first read the prompt. It's completely OK to think through the prompt, and even start working on it. As we start to write code, we may note something we missed before or realize that we need to ask a clarifying question. This is a normal part of the problem-solving process, so don't feel like we need to note everything or ask all our clarifying questions before we start working on the problem.
```

File: about-pses/example-pse.md [hunk 18]
```diff
@@ -29,64 +29,189 @@ Create a function named `winner` that takes two arguments, the move for `player_
...
+- ensure we are asking questions that will tell us new information about the problem space
+- check our understanding of the information we expect to get from those questions
+- uncover other observations that would help shape our approach and understand how they would affect our approach
+- uncover further questions that could be useful to ask and understand why those other questions could be helpful
+
+<br>
+
+For this question we will:
+1. Build a prompt using [the template linked here](https://gist.githubusercontent.com/ada-instructors/16c97dc4b16ab2bf449d9d7a81caeb16/raw/pse_observations_questions_review_template.md)
+2. Share the filled in prompt with an AI tool like ChatGPT
```
Comment (apradoada) [inline about-pses/example-pse.md:153]: Repeated comment from `highest-rated.md`:

```suggestion
2. Share the completed prompt with an AI tool like ChatGPT
```

File: about-pses/example-pse.md [hunk 19]
```diff
@@ -29,64 +29,189 @@ Create a function named `winner` that takes two arguments, the move for `player_
...
+3. After the initial review, ask *at least one* follow up question using the AI tool that furthers your understanding of the problem and why certain observations or questions are useful. Some examples could be asking questions to:
```
Comment (apradoada) [inline about-pses/example-pse.md:154]: Repeated comment from `highest-rated.md`:
```suggestion
3. After the initial review, ask the AI tool *at least one* follow up question that furthers your understanding of the problem and why certain observations or questions are useful. Some examples could be asking questions to: 
```

File: about-pses/example-pse.md [hunk 20]
```diff
@@ -173,46 +294,51 @@ def test_rock_beats_scissors():
...
 * id: 2bf3b413-e220-46aa-a99a-6c5f3d9715a9
 * title: Create Logical Steps
 * topics: pse
-
 ##### !question
 
-Without writing code, describe how you would implement `winner` in enough detail that someone else could write the code. 
-* It may be helpful to break up the problem/algorithm into smaller subproblems/algorithms. For example, 1. Handle invalid input, 2. Given valid input, perform the computation/solve the problem/etc.
-* Your logical steps could take the form of a numbered list, pseudo code, or anywhere in between. What's important at this stage is to think through and outline the implementation before writing code.
+Without writing code, describe how you would implement `winner` in enough detail that another Python developer could reasonably implement a solution. We should capture the main use cases, but the steps do not need to be a detailed plan for every contingency.
```
Comment (apradoada) [inline about-pses/example-pse.md:321]: Repeated comment from `highest-rated.md`:
```suggestion
Without writing code, describe how you would implement `winner` in enough detail that another developer could reasonably implement a solution. We should capture the main use cases, but the steps do not need to be a detailed plan for every contingency. 
```

File: about-pses/example-pse.md [hunk 21]
```diff
@@ -226,29 +352,84 @@ During an interview, we'd still want to be careful not to spend _too_ much time
...
+- the steps make sense for the problem being solved
+- the steps are not missing important steps or scenarios
+- the steps are agnostic of any particular language – steps should not include code syntax.
+- the steps are written with enough detail for another Python developer to understand how to create a solution
 
+<br>
+
+For this question we will:
+1. Build a prompt using [the template linked here](https://gist.githubusercontent.com/ada-instructors/670252696f1625cf0ed77c0997cd165d/raw/pse_logical_steps_review_template.md)
+2. Share the filled in prompt with an AI tool like ChatGPT
```
Comment (apradoada) [inline about-pses/example-pse.md:389]:
```suggestion
2. Share the completed prompt with an AI tool like ChatGPT
```

File: about-pses/example-pse.md [hunk 22]
```diff
@@ -274,19 +455,18 @@ def test_rock_beats_scissors():
     # assert
     assert result == "Player 1 wins!"
 
-def test_invalid_input():
+def test_winner_both_rock_returns_tie():
     # arrange
     player_1 = "rock"
-    player_2 = "lizards"
+    player_2 = "rock"
```
Comment (apradoada) [inline about-pses/example-pse.md:461]: Thank you for taking this test case out!

**Review by apradoada — COMMENTED** [review]
Overall I think this is looking really good! I just made some small adjustments that I thought might make certain sections flow better! Feel free to reach out if you have any questions!

File: 01/highest-rated.md [hunk 23]
```diff
@@ -176,44 +291,97 @@ def test_returns_none_with_zero_restaurants(self):
...
+
+##### !end-question
+##### !explanation
 
-<!-- ======================= END CHALLENGE ======================= -->
+As an example, let’s say we used the logical steps in the explanation for the question above in our prompt. Depending on exactly what ChatGPT shares, a reflection and summary might look like:
 
+<br>
+
+Chat Link: `<url to the conversation>`
```
Comment (yangashley) [inline 01/highest-rated.md:375]:
```suggestion
Chat Link: `<url to your conversation>`
```

File: 01/implement-highest-rated.checkpoint.md
Comment (yangashley) [inline 01/implement-highest-rated.checkpoint.md:1]: On lines 96-97 of the explanation, we show setting the first restaurant as the highest rated one and then looping from the beginning of the list to compare ratings, which works just fine. However, I remembered that this is what the ChatGPT conversation (in the Explanation section) calls out in Review Logical Steps: "I asked a follow up question and the response made me realize I should loop through the list starting from the second restaurant to avoid the redundancy of comparing the first item to itself."

Do we think it's fine to leave the solution like: 
```py
    highest_rated = restaurants[0]
    for restaurant in restaurants:
```

Or would we want to change it to:
```py
    highest_rated = restaurants[0]
    for restaurant in restaurants[1:]:
```

Or we could do for i in range.

File: about-pses/example-pse.md (hunk 17 above)
Comment (yangashley) [reply about-pses/example-pse.md:118]: This explanation is a rather long block of text and I feel like this is important to visually highlight this point. How do you feel about italicizing or bolding it?

```suggestion
We should keep in mind that sometimes, areas of clarification may not jump out at us when we first read the prompt. It's completely OK to think through the prompt, and even start working on it. **As we start to write code, we may note something we missed before or realize that we need to ask a clarifying question. This is a normal part of the problem-solving process. So don't feel like we need to note everything or ask all our clarifying questions before we start working on the problem.**
```

File: 01/highest-rated.md [hunk 24]
```diff
@@ -20,41 +20,156 @@ This function should take in a list of dictionaries named `restaurants` as a par
...
+
+<br>
+
+In the box below, list 5 or more observations about the problem or questions whose answers would clarify the problem statement. For each observation or question, include information on why that observation is important or why you are asking the question
+- For each observation, answer how that observation will affect your approach to the problem
+- For each question, describe what you are hoping to clarify about the problem and provide an answer which includes the effect your decision would have on how you might approach the problem.
+
+<br>
+
+As you come up with observations and questions, assume that error handling for invalid data is managed outside the function. We want to focus on the core behavior of the function we will write.
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply 01/highest-rated.md:45]: I think there is use to having those examples seen somewhere for folks who could use the clarification, but I don't know if it would be useful to repeat on every PSE. How do you feel about including it on the example PSE but leaving it off the core PSEs?

File: 01/highest-rated.md (hunk 14 above)
Comment (kelsey-steven-ada) [reply 01/highest-rated.md:58]: Good point, they have that info above and I'm sure they will be hearing it reinforced by us.

File: about-pses/about-pses.md (hunk 11 above)
Comment (kelsey-steven-ada) [reply about-pses/about-pses.md:19]: I like that, how do you feel about it being rephrased slightly as a second bullet point without further indentation?

```suggestion
- During an interview, writing some of the discussion points down can still be useful but our notes there would likely not be to the same level of detail as the responses for PSEs.
- Additionally, our PSEs will ask you to use AI tools to further explore and supplement your initial findings. You will likely not have access to AI tools in an Interview.
```

File: 01/highest-rated.md [hunk 25]
```diff
@@ -20,41 +20,156 @@ This function should take in a list of dictionaries named `restaurants` as a par
...
+- Did the AI tool suggest updates to the observations you made or questions you asked? 
+    - If so, what updates and why?
+- Did the AI tool suggest any new observations or questions?
+    - If so, what? Why would they be useful?
 
 ##### !end-hint
 ##### !explanation 
 
-Example clarifying questions:
+Everyone’s questions and conversation with ChatGPT will be a little different, and thus the summaries will look a little different. As an example, let’s say we used the questions and explanations from the "Explanation" section of the previous question to create our prompt.
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply 01/highest-rated.md:151]: How do you feel about including this on the example PSE to give folks more context, then leaving it off of the core PSEs?

File: 01/implement-highest-rated.checkpoint.md
Comment (kelsey-steven-ada) [reply 01/implement-highest-rated.checkpoint.md:1]: Good point, since it's noted in a response, we should use it in the implementation. Since I'm not sure how familiar folks will be with slicing at this point, I'm leaning towards `range`, but I'm good with either.

File: about-pses/example-pse.md (hunk 17 above)
Comment (kelsey-steven-ada) [reply about-pses/example-pse.md:118]: To help break up the chunk of text, how do y'all feel about moving that last couple sentences to a bullet point? On re-reading, I also didn't love telling people "Don't feel a certain way!", and rephrased it a little. 

```suggestion
We should keep in mind that sometimes, areas of clarification may not jump out at us when we first read the prompt. It's completely OK to think through the prompt, and even start working on it. As we start to write code, we may note something we missed before or realize that we need to ask a clarifying question. 
- This is a normal part of the problem-solving process! We don't need to note everything or ask all our clarifying questions before we start working on the problem.
```

File: about-pses/example-pse.md [hunk 26]
```diff
@@ -226,29 +352,84 @@ During an interview, we'd still want to be careful not to spend _too_ much time
...
+* id: eb41ea7a-cbdf-46d7-b48d-a03bebe872f7
+* title: Review Logical Steps
+* topics: pse
+##### !question
+
+We want to know if we are laying out an approach to the coding problem that makes sense for our context and if that approach is clearly conveying our thoughts on technical topics to others. Let’s once more use an AI tool like ChatGPT, this time to review the Logical Steps we wrote above. Our goals are to check if:
+- the steps make sense for the problem being solved
+- the steps are not missing important steps or scenarios
+- the steps are agnostic of any particular language – steps should not include code syntax.
+- the steps are written with enough detail for another Python developer to understand how to create a solution
```
Comment (kelsey-steven-ada) [inline about-pses/example-pse.md:383]:
```suggestion
- the steps are written with enough detail for another developer to understand how to create a solution
```

File: about-pses/example-pse.md (hunk 17 above)
Comment (yangashley) [reply about-pses/example-pse.md:117]: Making it a bullet point looks good to me!

File: 01/highest-rated.md (hunk 25 above)
[the remark this answers is not in the document]
Comment (yangashley) [reply 01/highest-rated.md:150]: I think it's ok to just have it on the example PSE!

File: about-pses/example-pse.md [hunk 27]
```diff
@@ -29,64 +29,188 @@ Create a function named `winner` that takes two arguments, the move for `player_
...
+- Did the AI tool uncover anything about the questions you asked that you hadn’t considered?
+- Did the AI tool suggest updates to the observations you made or questions you asked? 
+    - If so, what updates and why?
+- Did the AI tool suggest any new observations or questions?
+    - If so, what? Why would they be useful?
+
+##### !end-hint
+##### !explanation 
+
+Everyone’s observations & questions, and conversation with ChatGPT will be a little different, thus, the summaries will look a little different. As an example, let’s say we used the sample response from "Describe Your Understanding" above to fill in the prompt template.
```
Comment (kelsey-steven-ada) [inline about-pses/example-pse.md:186]:
```suggestion
Everyone’s observations & questions, as well as their conversations with ChatGPT. will be a little different. Because of this, the summaries created will look a little different.

After you submit a response in Learn, you'll see "Show Explanation" appear and when it is clicked an explanation is revealed. For this example, we'll use the questions and explanations from the "Explanation" section of the previous question as examples to create our prompt. 
```

**Review by yangashley — APPROVED** [review]
